sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp

```diff
@@ -63,7 +63,7 @@ namespace Azure { namespace Core { namespace _internal {
      *
      * @param other Another extendable enumeration to be compared.
      */
-    bool operator==(T const& other) const noexcept
+    bool operator==(ExtendableEnumeration<T> const& other) const noexcept
```

Comment:
@antkmsft can you please help me understand why this change was necessary and the implications?

Even though this type is internal, is this considered a breaking change (where a previous call to `==` with different types of "arguments" now fail to compile?